### ISSUE TITLE
Plugin broken on Linux, even after modifying its code myself.

### ISSUE THREAD

**Issue by ScalSaver** [tgiesselmann/gimp-stable-boy#17]
Hello ! I'm the complaining Redditor, /u/SeventhDeadlyBless.

So. 

Specs :

Linux Mint 21, 64 bits.
Conda environement, [link to environement.yml](https://pastebin.com/ZMJnJBZz)
Can give more detail if asked : I'm my own sysadmin.

test shell command :
` gimp --verbose --console-messages`

```
Querying plug-in: '/usr/lib/gimp/2.0/plug-ins/gimp_stable_boy.py'
Traceback (most recent call last):
  File "/usr/lib/gimp/2.0/plug-ins/gimp_stable_boy.py", line 45, in <module>
    from gimp_params import GIMP_PARAMS, IMAGE_TARGETS as IMG_TARGET, SAMPLERS, UPSCALERS
  File "/usr/lib/gimp/2.0/plug-ins/gimp_params.py", line 36, in <module>
    (gimpfu.PF_STRING, 'prompt', 'Prompt', ''),
NameError: name 'gimpfu' is not defined
```

I put the imports of gimpfu in comment, as I can't find it for pip, and uncommenting it results in an error message as if I haven't it installed. "Gimpfu not found", to wit. It's an API shipped with Gimp, it **has to** be there and in my PATH.

I thought the module pgimp was the python binding for the aformentionned API, but a couple of issues : 
 1. This module is very freaking old. As in "unmaintained for more than a whole decade" old. So most of its purpose could be lost, and worse : its documentation apparently disappeared too. Only remains a very [unhelpful and sparse documentation website.](https://pgimp.readthedocs.io/en/latest/) It's bad.
 1. I've seen the between 3-4 different names for gimp APIs. Which one pgimp supports ? Fuck if we can know, now. Which binding goes with which API ? [This antiquity](https://www.gimp.org/docs/python/) is my best source on the matter. It's dependency hell.
 1. Every other source says Gimp manages only Python2.7, but pgimp installs only on Python3.9. Why ? who knows anymore.  

A little tidbit about why I updated/upgraded the libraries : some aren't available to me, even on python2.7. Updating to Python3.9 seemed rational, as pgimp seemed to want to be of help.

 How any of this even worked in the first place ? I'm baffled at every turn in this labyrinth.
 
 PS : Here is my latest modified copies of your source files. If you manage to pick up on anything my modifications wreaked ... There has to be a couple of things. I'm a messy software editor.

[gimp_params.py.txt](https://github.com/tgiesselmann/gimp-stable-boy/files/10295597/gimp_params.py.txt)
[gimp_stable_boy.py.txt](https://github.com/tgiesselmann/gimp-stable-boy/files/10295613/gimp_stable_boy.py.txt)

**Comment by infernix** [comment]
This is a rabbithole you won't get out of (python3 support).

Best solution is to just use the GIMP Flatpak from https://www.gimp.org/downloads/. Works OOTB then.

**Comment by ScalSaver** [comment]
Like installing the flatpack, putting the .py in the right place, and enjoy infinite AI image generation ?

I feel like it's caving to the warning Mint devs gave us about snapd. Loose-loose / "infinite dependency hell-software ease for safety" situation.

But i am at the point there's only doing as you advise left to try.

**Comment by thndrbrrr** [comment]
@ScalSaver Any success with the flatpack install?

**Comment by seefood** [comment]
just got around to installing, had the same issue, flatpack is showing Stable Boy in the menu, but all under it are greyed, including preferences. also this is not a great solution, having to keep two copies of about 1GB each on the system, each with different plugins and all. I want one copy that does everything.

**Comment by thndrbrrr** [comment]
Agreed, @seefood, this is not ideal. 

As for why the menu items are greyed out – you probably didn't have an image open? The menu is only active then. (Which makes sense, except for the preferences ... hmmm.)

**Comment by ScalSaver** [comment]
Hello.

Replying to state I can't get my computer setup in testing condition :

1. Couldn't get the plugin to run, on neither installation of GIMP.
Flatpack seem inaccessible, bottled up. Managed to scavenge some
python-gimp from it once, only to never find the place ever again. The
settings and files I've scrapped in aren't functionnal.

2. I'd try more if my booting stack wasn't posting through by the smallest
thread imaginable. Issues between mobo bios and grub/initramfs/actual linux
kernel. No clue where exactly, can't roll back the bios. Stuck with older
cpu microcode and kernel, until that's resolved. It's like working on top
of a tower of cards until this issue persists.

I'm off the race. I can handle only one of those issues.
